vaadin-date-picker-flow-parent/vaadin-date-picker-testbench/src/main/java/com/vaadin/flow/component/datepicker/testbench/DatePickerElement.java

```diff
@@ -154,14 +155,18 @@ protected String getValue() {
     }
 
     /**
-     * Opens the overlay, sets the value to the inner input element as a string
-     * and closes the overlay. This simulates the user typing into the input and
-     * triggering an update of the value property.
+     * Emulates the user changing the value. This in practice means clearing
+     * {@code value} of the {@code input} element and then setting the new value
+     * by sending the respective characters + Enter via the Selenium
+     * {@code sendKeys} API.
+     *
+     * @param string
+     *            the value to set
      */
     public void setInputValue(String value) {
-        this.open();
-        setProperty("_inputValue", value);
-        this.close();
+        TestBenchElement input = $("input").first();
+        input.setProperty("value", "");
```

Comment:
As I found, when calling this method with an empty string, no `input` event will be fired with the current approach of clearing the value.

Reply:
Fixed by https://github.com/vaadin/flow-components/pull/3576/commits/7657174dcc69ac13aea9324cd786e2d4cd05ded7